src/main/java/org/opensearch/knn/index/codec/util/KNNVectorSerializer.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.codec.util;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+/**
+ * Interface abstracts the vector serializer object that is responsible for serialization and de-serialization of k-NN vector
+ */
+public interface KNNVectorSerializer {
+    /**
+     * Serializes array of floats to array of bytes
+     * @param input array that will be converted
+     * @return array of bytes that contains serialized input array
+     * @throws Exception
+     */
+    byte[] floatToByteArray(float[] input) throws Exception;
+
+    /**
+     * Deserializes all bytes from the stream to array of floats
+     * @param byteStream stream of bytes that will be used for deserialization to array of floats
+     * @return array of floats deserialized from the stream
+     * @throws IOException
+     * @throws ClassNotFoundException
```

Comment:
Do we need to have ClassNotFoundException here?

Reply:
It comes from the serializer implementation, ObjectInputStream.readObject(). 
I've dig deeper into exceptions and seems we just catch and re-throw RuntimeException at the higher level, so I moved this logic inside the serializer and removed all exceptions from the interface method signature. I hope it makes sense